### PR TITLE
install.sh: Fix -h handling and error on unrecognized parameters

### PR DESCRIPTION
**Pull request by CyberShadow**

### CODE REVIEW

File: script/install.sh [hunk 1]
```diff
@@ -277,6 +277,11 @@ parse_args() {
             dmd | dmd-* | gdc | gdc-* | ldc | ldc-*)
                 compiler=$1
                 ;;
+
+            *)
+                usage
+                fatal "Unrecognized command-line parameter: $1"
```
Comment (wilzbach) [inline script/install.sh]: Maybe first showing the error message first?

Comment (CyberShadow) [reply script/install.sh]: No. I intentionally put it after the usage text. You're more likely to notice stuff at the bottom of your terminal window (near the prompt) than near the top.

Comment (wilzbach) [reply script/install.sh]: Fair enough, I wouldn't have minded anyways

Comment (MartinNowak) [reply script/install.sh]: Indeed, it's weird that most tools do it vice versa.

**Comment by jondegenhardt** [comment]
This PR appears to have exposed a bug in the install script. The argument parsing for `-p | --path` (https://github.com/dlang/installer/blob/master/script/install.sh#L254-L259) take a second argument, the path. However, the '-p|--path' argument handler doesn't actually consume the argument, which then get run through arg processing loop. Previously it was ignored. Now the script exits.

I found this trying to run the docker-ldc script, uses the -p option: https://github.com/lindt/docker-ldc/blob/master/Dockerfile#L11

**Comment by CyberShadow** [comment]
Looks like Martin had fixed this by the time you ran into it: 143bbf8669447f66f51402ac6db90de1350407ca

**Comment by MartinNowak** [comment]
Was already wondering how that could have been there forever ;).

**Comment by jondegenhardt** [comment]
And thanks for putting the error message at the bottom of the window!

**Pull request merged**